File: js/1013.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const T = +inputs[0];

const regex = /^((100+1+)|01)+$/;
const result = [];

for (let i = 1; i <= T; i++) {
  const s = inputs[i];

  result.push(regex.test(s) ? 'YES' : 'NO');
}

console.log(result.join('\n'));

File: c++/1013.cpp
#include <iostream>
#include <regex>
#include <string>

using namespace std;

int main() {
  freopen("example.txt", "r", stdin);

  int N;
  cin >> N;

  regex re("^((100+1+)|01)+$");

  for (int i = 0; i < N; i++) {
    string s;
    cin >> s;

    if (regex_match(s, re))
      cout << "YES";
    else
      cout << "NO";

    cout << '\n';
  }

  return 0;
}